firestore/integration_test_internal/src/jni/arena_ref_test.cc

```diff
@@ -0,0 +1,278 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "firestore/src/jni/arena_ref.h"
+#include "firestore/src/android/firestore_android.h"
+#include "firestore/src/common/make_unique.h"
+#include "firestore/src/jni/loader.h"
+#include "firestore/src/jni/long.h"
+
+#include "firestore_integration_test.h"
+
+#include "gtest/gtest.h"
+
+namespace firebase {
+namespace firestore {
+namespace jni {
+namespace {
+
+constexpr char kException[] = "java/lang/Exception";
+Method<String> kConstructor("<init>", "(Ljava/lang/String;)V");
+
+class ArenaRefTestAndroid : public FirestoreIntegrationTest {
+ public:
+  ArenaRefTestAndroid() : loader_(app()), env_(make_unique<Env>(GetEnv())) {
+    loader_.LoadClass(kException);
+    loader_.Load(kConstructor);
+  }
+
+  ~ArenaRefTestAndroid() override {
+    // Ensure that after the test is done that any pending exception is cleared
+    // to prevent spurious errors in the teardown of FirestoreIntegrationTest.
+    env_->ExceptionClear();
+  }
+
+  Env& env() const { return *env_; }
+
+  void throwException() {
+    Local<Class> clazz = env().FindClass(kException);
+    jmethodID ctor =
+        env().GetMethodId(clazz, "<init>", "(Ljava/lang/String;)V");
+
+    Local<String> message = env().NewStringUtf("Testing throw");
```

Comment:
I think you can just get rid of these throwException() and clearExceptionOccurred() helper methods.

throwException() can be replaced by 
```cpp
env.Throw(CreateException(env, "Test Exception"));
```

and clearExceptionOccurred() can be replaced by
```cpp
env.ExceptionClear();
```

Admittedly, this leads to a tiny bit of code duplciation between test cases but that's acceptable because, IMO, it improves readabililty of the test cases. 

See go/tott/598 about "DAMP" tests rather than the "DRY" principle that is common in non-testing code.